xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp

```diff
@@ -330,19 +289,38 @@ void CAddonVideoCodec::Reset()
   {
     if (ret == VIDEOCODEC_RETVAL::VC_PICTURE)
     {
-      m_bufferPool->ReleaseBuffer(m_lastPictureBuffer);
+      videoBuffer = m_map[m_lastPictureBuffer];
+      if (videoBuffer)
+        videoBuffer->Release();
+      m_map.erase(m_lastPictureBuffer);
       m_lastPictureBuffer = picture.decodedData;
     }
   }
-  m_bufferPool->ReleaseBuffer(m_lastPictureBuffer);
+  videoBuffer = m_map[m_lastPictureBuffer];
+  if (videoBuffer)
+    videoBuffer->Release();
+  m_map.erase(m_lastPictureBuffer);
   m_lastPictureBuffer = nullptr;
 
   m_struct.toAddon.reset(&m_struct);
 }
 
 bool CAddonVideoCodec::GetFrameBuffer(VIDEOCODEC_PICTURE &picture)
 {
-  return m_bufferPool->GetBuffer(picture);
+  CVideoBuffer *videoBuffer = m_videoBufferManager.Get(AV_PIX_FMT_YUV420P, picture.decodedDataSize);
```

Comment:
I see, why don't you drop Get(AVPixelFormat format, int width, int height)? clieant should call Get(fmt, width * height) instead